src/effects/effectknobparameterslot.h

```diff
@@ -49,7 +49,7 @@ class EffectKnobParameterSlot : public EffectParameterSlotBase {
         return QString("EffectKnobParameterSlot(%1,%2)").arg(m_group).arg(m_iParameterSlotNumber);
     }
 
-    SoftTakeover* m_pMetaknobSoftTakeover;
+    SoftTakeover m_metaknobSoftTakeover{};
```

Comment:
why?
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-in-class-initializer

Reply:
In this case `SoftTakeover` is automatically initialized with its default constructor, the {} is redundant. 

In general we have the rule to prefer the constructor member initializer list:
https://github.com/mixxxdj/mixxx/wiki/Coding-Guidelines#non-static-data-member-initializers

In addition this practice avoids a static inline constructor call and minimizes recompilation in case the initializer changes and redundant initializers.

I just noticed there are also some pending issues: 
https://github.com/isocpp/CppCoreGuidelines/issues/2213
https://github.com/isocpp/CppCoreGuidelines/issues/2032

The real issue in this case is that we may forget to initialize all members. I am afraid we have not set the compiler/linter to warn us about that. We should consider to set it up.

Reply:
> In this case `SoftTakeover` is automatically initialized with its default constructor, the {} is redundant.

I am aware that this applies for class types, it does not however apply to primitive types so I wanted to establish this style regardless of what type it is in an effort to simplify the code...

Reply:
from what I could tell, `-Weffc++` does the trick. Unfortunately it flags lots of code, so enabling that will have to be a separate PR.